sdk/storage/azure-storage-blob/v10_v12.md

```diff
@@ -0,0 +1,437 @@
+```java
+// Initialize the credentials
+SharedKeyCredentials credential = new SharedKeyCredentials("${accountName}", "${accountKey}");
+// How service object take in the credentials.
+BlockBlobURL blockBlobURL = new BlockBlobURL(new URL("<your-storage-blob-url>"), StorageURL.createPipeline(credential, new PipelineOptions()));
+```
+
+Since the builder in v12 manages pipeline generation, you can just hand the `SharedKeyCredential` to the builder.
+```java
+// Initialize the credentials
+SharedKeyCredential credential = new SharedKeyCredential("${accountName}", "${accountKey}");
+// How service object take in the credentials.
+BlobServiceClient client = new BlobServiceClientBuilder()
+    .endpoint("https://" + accountName + ".blob.core.windows.net")
+    .credential(credential)
+    .buildBlobServiceClient();
+```
+
+### SASToken
+A URL with a SAS token looks like the following: `https://${accountName}.blob.core.windows.net/?${sasToken}`.
+Refer the [documentation](https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-file_12.0.0-preview.3/sdk/storage/azure-storage-blob/README.md) for more info.
+
+See section [Generate SAS token](#generate-sas-token) for how to generate the token programatically.
+
+In v10/v11, you had to provide the URL with the token attached, which often resulted in URL manipulation when a token could be used on more than one resource.
+```java
+// Here is how the blob object take in the sas token generated in above section.
+BlobURLParts parts = URLParser.parse(cu.createBlobURL(blobName).toURL())
+parts.withSasQueryParameters(sasToken).withScheme("https")
+BlockBlobURL bu = new BlockBlobURL(parts.toURL(), StorageURL.createPipeline(
+    new AnonymousCredentials(), new PipelineOptions()))
+```
+
+In v12, SAS tokens can be managed just like any other credential.
+```java
+// Initialize the credentials
+SASTokenCredential credential = SASTokenCredential.fromSASTokenString("${sasToken}");
+// How service object take in the credentials.
+BlobServiceClient client = new BlobServiceClientBuilder()
+    .endpoint("https://${accountName}.blob.core.windows.net")
+    .credential(credential)
+    .buildBlobServiceClient();
+```
+
+However, if you already have the full URL with SAS token attached, you can use that too:
+```java
+BlobServiceClient client = new BlobServiceClientBuilder()
+    .endpoint("https://${accountName}.blob.core.windows.net/?${sasToken}")
+    .buildBlobServiceClient();
+```
+
+## Provide side by side sync-over-async APIs
+Each type (Container, Blob, etc.) has both a sync and async client. It is easy and intuitive to use whichever one is preferable in a given context without having to bother about clutter from the other.
+Suppose the Container Client been initialized as `${containerClient}`. Please refer the [Build Client](#build-client) section.
+
+* Example of using Async API in V12. 
+```java
+Mono<BlobProperties> response = containerClient.getProperties();
```

Comment:
I think we should make it clear that these are two different types, so line 193 should read "Suppose the ContainerClient has been initialized as `containerClient` and the ContainerAsyncClient has been initialized as `containerAsyncClient` and then update the variable on this line to be containerAsyncClient.

I also think this should maybe get assigned to propertiesMono or something since we have an actual type called Response that is not being used here.